tests/interfaces/records-write.spec.ts

```diff
@@ -287,13 +316,48 @@ describe('RecordsWrite', () => {
       const write = await RecordsWrite.createFrom({
         recordsWriteMessage : recordsWrite.message,
         datePublished       : getCurrentTimeInHighPrecision(),
-        authorizationSigner : Jws.createSigner(author)
+        signer              : Jws.createSigner(author)
       });
 
       expect(write.message.descriptor.published).to.be.true;
     });
   });
 
+  describe('parse()', () => {
+    it('should throw if delegated grant is given but signer is not given', async () => {
+      const alice = await TestDataGenerator.generatePersona();
+      const bob = await TestDataGenerator.generatePersona();
+
+      const scope: PermissionScope = {
+        interface : DwnInterfaceName.Records,
+        method    : DwnMethodName.Write
+      };
+      const grantToBob = await PermissionsGrant.create({
+        delegated           : true, // this is a delegated grant
+        dateExpires         : createOffsetTimestamp({ seconds: 100 }),
+        description         : 'Allow to Bob write as me in chat protocol',
+        grantedBy           : alice.did,
+        grantedTo           : bob.did,
+        grantedFor          : alice.did,
+        scope,
+        authorizationSigner : Jws.createSigner(alice)
+      });
+
+      const recordsWrite = await RecordsWrite.create({
+        signer         : Jws.createSigner(alice),
+        delegatedGrant : grantToBob.asDelegatedGrant(),
+        dataFormat     : 'application/octet-stream',
+        data           : TestDataGenerator.randomBytes(10),
+      });
+
+      delete recordsWrite.message.authorization!.authorDelegatedGrant; // intentionally remove `authorDelegatedGrant`
```

Comment:
Sanity: This looks to be the opposite of what is described in the test name `should throw if delegated grant is given but signer is not given`.

Reply:
Ooops, thanks for the keen eyes, C&P error (there is another test with the exact same name), fixed!